Content.Server/GameObjects/Components/Damage/DamageOnToolInteractComponent.cs

```diff
@@ -0,0 +1,108 @@
+
+        /* Set in YAML */
+        protected int Damage;
+        protected bool UsingWelding;
+        protected bool UsingAnchoring;
+        protected bool UsingCutting;
+        protected bool UsingMultitool;
+        protected bool UsingPrying;
+        protected bool UsingScrewing;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref Damage,            "damage",           0);
+            serializer.DataField(ref UsingWelding,      "usingWelding",      false);
+            serializer.DataField(ref UsingAnchoring,    "usingAnchoring",   false);
+            serializer.DataField(ref UsingCutting,      "usingCutting",     false);
+            serializer.DataField(ref UsingMultitool,    "usingMultitool",   false);
+            serializer.DataField(ref UsingPrying,       "usingPrying",      false);
+            serializer.DataField(ref UsingScrewing,     "usingScrewing",    false);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _actSystem = _entitySystemManager.GetEntitySystem<ActSystem>();
+        }
+
+            public bool InteractUsing(InteractUsingEventArgs eventArgs)
+        {
+
+            //Used to support multiple tool interactions.
+            //If two tools were supported then the first check would return false and be past straight to return breaking function and second check would never be carried out.
+            //assumes interaction can only be with one tool at one time.
+            bool result = false; 
+
+            if (UsingWelding)
+            {
+                if (eventArgs.Using.HasComponent<WelderComponent>())
+                {
+                    WelderComponent tool = eventArgs.Using.GetComponent<WelderComponent>();
+                    if(tool.WelderLit)
+                    if (tool.HasQuality(ToolQuality.Welding))
+                    result = CallDamage(eventArgs);
+                }
+                if (result is true) return true;
+            }
+
+            if (UsingAnchoring) { result = HandleToolInteract(eventArgs, ToolQuality.Anchoring); if (result is true) return true; }
+
+            if (UsingCutting) { result = HandleToolInteract(eventArgs, ToolQuality.Cutting); if (result is true) return true; }
+
+            if (UsingMultitool) { result = HandleToolInteract(eventArgs, ToolQuality.Multitool); if (result is true) return true; }
+
+            if (UsingPrying) { result = HandleToolInteract(eventArgs, ToolQuality.Prying); if (result is true) return true; }
+
+            if (UsingScrewing) { result = HandleToolInteract(eventArgs, ToolQuality.Screwing); if (result is true) return true; }
+
+            return result;
+        }
+
+        protected bool HandleToolInteract(InteractUsingEventArgs eventArgs, ToolQuality quality)
+        {
+        if (eventArgs.Using.HasComponent<ToolComponent>())
```

Comment:
Your indentation is incorrect here.

Reply:
Would you also like me to expand the if statements at 68-76 or leave them as single lines?

Reply:
It's up to you - I don't think there's a codebase standard.